### ISSUE TITLE
sourceInstanceName filtering

### ISSUE THREAD

**Issue by Ehesp** [ChristopherBiscardi/gatsby-mdx#265]
Hey

I've got a project where I clone a Github repository locally. One of my directories in this clone (`/docs`) contains all of my `.md` files which I'm using to generate pages in Gatsby.

I've got it all working, but I also need to use the `gatsby-source-filesystem` plugin to work with other files in the clone, separate from the documentation directory. There are other `.md` files within the rest of the project (README, CONTRIBUTING etc) which are currently getting picked up bt gatsby-mdx, and error because they've got some stuff unrelated to mdx parsing.

Right now, the plugin checks for a File parent type and then goes to create a node... Is it possible (or some other way) to do the following?

```
// gatsby-config
	{
      resolve: 'gatsby-mdx',
      options: {
        extensions: ['.md'],
		source: 'foobar',
      }
    },
	{
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'foobar',
        path: `${__dirname}/.github-clone/docs`,
      },
    },
	{
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'barbaz',
        path: `${__dirname}/.github-clone`,
		ignore: [
			`${__dirname}/docs`,
		]
      },
    },
```

Basically, I'd then be able to specify a `source`, which is set on `gatsby-source-filesystem` as `sourceInstanceName` on the parent?

Happy to submit a PR unless there's a better way of doing it - assuming it'd just be a check [here](https://github.com/ChristopherBiscardi/gatsby-mdx/blob/master/packages/gatsby-mdx/gatsby/on-create-node.js#L41)?

**Comment by ChristopherBiscardi** [comment]
@Ehesp IMO this is a core gatsby issue. We need a way to be able to target transfomers to specific groupings of nodes rather than media types.

 I'd be open to introducing a blocklist config function in gatsby-mdx in the meantime such that we could include it as the first check in the list for processing at the location you've specified already... and I won't want to document it to avoid people using it when there's better options. It is also likely to break if this functionality *does* make it into gatsby core at some point.

```ja
{
  shouldBlockNodeFromTransformation: node => {} // return boolean
}
```

@chaseadamsio you can also do that in your graphql query without the additional processing by accessing the sourceInstanceName on the parent. The inline fragment method defined below is more extensible if you ever want to use a headless CMS in the future or if you have other markdown media types in your project (like react-docgen generates without a file node).

```graphql
{
  allMdx {
    edges {
      node {
        id
        parent {
          ... on File {
            sourceInstanceName
          }
        }
      }
    }
  }
}
```

**Comment by Ehesp** [comment]
@ChristopherBiscardi Yeah agreed. I submitted a PR to include it, nothing fancy but should help somewhat!

**Comment by sanderfish** [comment]
@ChristopherBiscardi With that last query, would it be possible to filter on `sourceInstanceName` (in a template)?

I'm using gatsby-mdx and gatsby-source-filesystem to create `posts` and `pages`, but when using `allMdx` I'm unable to filter for just one of these – my queries return both `posts` and `pages`.

**Comment by eeyorr** [comment]
@sanderfish you can filter on the fileAbsolutePath:

`allMdx(filter: {fileAbsolutePath: {regex: "\\/posts/"}}) {`
